### PR TITLE
bpo-29970: Make ssh_handshake_timeout None by default

### PR DESCRIPTION
**Pull request by asvetlov**
Float default prevents checking for passing `ssh_handshake_timeout` with plain socket connection.

Raise `ValueError` if `ssh_handshake_timeout` is specified without `ssl`.

Postfix for #4825

https://bugs.python.org/issue29970

### CODE REVIEW

File: Lib/asyncio/base_events.py [hunk 1]
```diff
@@ -674,6 +673,10 @@ def _getaddrinfo_debug(self, host, port, family, type, proto, flags):
                                  'when using ssl without a host')
             server_hostname = host
 
+        if ssl_handshake_timeout is not None and not ssl:
+            raise ValueError(
+                'ssl_handshake_timeout is only meaningfu with ssl')
```
Comment (1st1) [inline Lib/asyncio/base_events.py]: meaningfu -> meaningful

**Comment by 1st1** [comment]
BTW, what happens if someone passes `ssl_handshake_timeout=0.0`?  Do we need to explicitly prohibit this?

File: Lib/asyncio/base_events.py (hunk 1 above)
Comment (asvetlov) [reply Lib/asyncio/base_events.py]: Done

**Comment by asvetlov** [comment]
zero and negative timeout is an error but I doubt if we should prohibit it: there still is a possibility to pass `1e-30` which is obviously wrong.

**Comment by 1st1** [comment]
> zero and negative timeout is an error but I doubt if we should prohibit it: there still is a possibility to pass 1e-30 which is obviously wrong.

In many APIs, passing `0.0` disables the mechanism, and that's why we should either disable the timeout, or raise an error.  Otherwise it's just confusing.

**Comment by 1st1** [comment]
I vote for raising an error for `0.0`.  There's no point in disabling the timeout.

If someone passes `1e-10`, it's them clearly wanting to get a `TimeoutError`, that's it.

**Comment by asvetlov** [comment]
Ok, will add a code for raising

**Comment by asvetlov** [comment]
Done

**Review by 1st1 — APPROVED** [review]

File: Lib/test/test_asyncio/test_sslproto.py [hunk 2]
```diff
@@ -75,6 +75,22 @@ def test_handshake_timeout(self):
             self.loop.run_until_complete(tasks.sleep(0.2, loop=self.loop))
         self.assertTrue(transport.abort.called)
 
+    def test_handshake_timeout_zero(self):
+        sslcontext = test_utils.dummy_ssl_context()
+        app_proto = mock.Mock()
+        waiter = mock.Mock()
+        with self.assertRaises(ValueError):
```
Comment (1st1) [inline Lib/test/test_asyncio/test_sslproto.py]: assertRaisesRegex

**Comment by 1st1** [comment]
Wait, please use `assertRaisesRegex` to validate part of the error message.  `ValueError` can be raised from anywhere, and sometimes `assertRaises` tests something entirely different.

**Pull request merged**